src/main/java/frc/robot/trajectory/PurePursuit.java

```diff
@@ -0,0 +1,107 @@
+package frc.robot.trajectory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import frc.robot.systems.DriveFSMSystem;
+
+public class PurePursuit {
+
+	private DriveFSMSystem fsmSystem;
+
+	private ArrayList<Point> keyPoints = new ArrayList<>();
+	private ArrayList<Point> pathPoints = new ArrayList<>();
+	private int lastClosestPointIndex = 0;
+	// lookahead point is this many points ahead of the closest point
+	private final int lookaheadDistance = 6;
+
+	// in inches
+	private static final double SPACING = 6.0;
+
+
+	public PurePursuit(ArrayList<Point> initialPoints, DriveFSMSystem fsmSystem) {
```

Comment:
I'm not a fan of passing in fsmSystem to the constructor. DriveFSMSystem owns the PurePursuit instance, not the other way around. As far as I can tell, we only use fsmSystem to query values like getRobotPosArc and getheading. These should be passed in the method parameters when DriveFSMSystem calls the method.